tap_google_analytics/client.py

```diff
@@ -13,12 +13,36 @@
 
 LOGGER = singer.get_logger()
 
+def is_retryable_403(e):
+    response = e.response
+    if not _is_json(response):
+        return False
+
+    retryable_errors = {"userRateLimitExceeded", "rateLimitExceeded", "quotaExceeded"}
+    error_reasons = {error.get('reason') for error in response.json().get('errors',[])}
+
+    if any(error_reasons.intersection(retryable_errors)):
+        return True
+
+    return False
+
 def should_giveup(e):
-    if e.response.status_code == 429:
+    """
+    Note: Due to `backoff` expecting a `giveup` parameter, this function returns:
+
+    True - if the exception is NOT retryable
+    False - if the exception IS retryable
+    """
+    should_retry = e.response.status_code == 429 or is_retryable_403(e)
+
+    if should_retry and _is_json(e.response):
```

Comment:
Why are you calling is_json here if that check is already part of the is_retryable_403 method?

Reply:
Mostly for extra safety in the case of a 429, in case google decides to return a 429 with an XML body or something.

Reply:
Right but isn't this method being called twice? Like if is_retryable_403 already calls is_json than is_retryable_403 returns false... Why call is_json again? It just seems like you could re-organize the boolean logic you have here to avoid calling it twice.